src/prototype.rs

```diff
@@ -1322,11 +1357,24 @@ impl<'a> InlineScanner<'a> {
 
     fn to_node_and_ix(&self) -> (usize, usize) {
         let mut cur = self.cur;
-        if self.tree.nodes[cur].item.end == self.ix {
+        if cur != NIL && self.tree.nodes[cur].item.end == self.ix {
```

Comment:
Why the extra condition?

Reply:
Because it's quite possible for `cur` to be `NIL` here if the client has scanned all the way to the end. I'm pretty sure if you revert this you can see a panic on one of the tests.